packages/shared/ReactComponentStackFrame.js

```diff
@@ -33,17 +34,26 @@ export function describeBuiltInComponentFrame(name: string): string {
       } catch (x) {
         const match = x.stack.trim().match(/\n( *(at )?)/);
         prefix = (match && match[1]) || '';
+        suffix =
+          x.stack.indexOf('\n    at') > -1
+            ? // V8
+              ' (<anonymous>)'
+            : // JSC/Spidermonkey
+            x.stack.indexOf('@') > -1
+            ? '@unknown:0:0'
+            : // Other
+              '';
       }
     }
     // We use the prefix to ensure our stacks line up with native stack frames.
-    return '\n' + prefix + name;
+    return '\n' + prefix + name + suffix;
   } else {
     return describeComponentFrame(name);
   }
 }
 
 export function describeDebugInfoFrame(name: string, env: ?string): string {
-  return describeBuiltInComponentFrame(name + (env ? ' (' + env + ')' : ''));
+  return describeBuiltInComponentFrame(name + (env ? ' [' + env + ']' : ''));
```

Comment:
This was my original reasoning for parenthesis https://github.com/facebook/react/pull/28415#issuecomment-1961668743

However, after further research it seems like e.g. `(native)` is no longer used by V8 and the only non-location is `(<anonymous>)` so this pattern doesn't follow and it actually gets parsed as if it's a real location.

Meanwhile brackets doesn't seem to have any meaning in any format as far as I can tell from further research so just becomes a name.

All formats are ambiguous when the actual name contains meaningful punctuations like `(` or `@`.